### PR TITLE
Implement polls edit and delete.

### PR DESCRIPTION
**Pull request by gyakkun**
Now the button **Edit** and **Delete** in Staff/polls page should work.

Add necessary language items in resources/lang for zh-CN, zh-TW and en.

( Note: As I just begin writing php this week, the code may lack exception handling and so on. **Serious review is needed.** )

### CODE REVIEW

File: resources/views/Staff/poll/edit.blade.php [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+@section('content')
+    <div class="container">
+        <div class="row">
+            <div class="page-title">
+                <h1>@lang('poll.edit-poll')</h1>
+            </div>
+            <div class="col-md-12">
+                <div class="panel panel-chat">
+                    <div class="panel-heading">
+                        @lang('common.edit')
```
Comment (gyakkun) [inline resources/views/Staff/poll/edit.blade.php]: Oh these 3 lines should be `@lang('poll-edit')`.  I'm localizing the Staff/* views. @lang(trans_choice('common.a-an-art',false)) is used for switching a/an in English.

File: app/Http/Controllers/Staff/PollController.php [hunk 2]
```diff
@@ -103,4 +103,97 @@ public function store(StorePoll $request)
...
+    /**
+     * Update A New Poll.
+     *
+     * @param StorePoll $request
+     *
+     * @return Illuminate\Http\RedirectResponse
+     */
+    public function update(StorePoll $request, $id)
+    {
+        $poll = Poll::findOrFail($id);
```
Comment (gyakkun) [inline app/Http/Controllers/Staff/PollController.php]: Lack the ip_checking and multiple_choice flag.

**Comment by HDVinnie** [comment]
@gyakkun Im going to review this tonight when home. I do think however this whole polls system needs to be re-written from scratch which I plan to do. It was poorly designed and made. Possible refactor might suffice. Ill keep you updated. Thanks again for the contributions and look forward to your translation (i18n) pull requests.

File: app/Http/Controllers/Staff/PollController.php (hunk 2 above)
Comment (gyakkun) [reply app/Http/Controllers/Staff/PollController.php]: 6ede4d0287a1f2ae65a3a81c50949c6652ab3d71

File: resources/views/Staff/poll/edit.blade.php (hunk 1 above)
Comment (gyakkun) [reply resources/views/Staff/poll/edit.blade.php]: 2643fa6f8714e732390668e74b8754b537e384c6

File: resources/views/Staff/poll/index.blade.php [hunk 3]
```diff
@@ -40,8 +40,17 @@
                             <td><a href="{{ url('/dashboard/polls/' . $poll->id) }}">{{ $poll->title }}</a></td>
                             <td>{{ date('d M Y', $poll->created_at->getTimestamp()) }}</td>
                             <td>
-                                <a href="#" class="btn btn-warning">@lang('common.edit')</a>
-                                <a href="#" class="btn btn-danger">@lang('common.delete')</a>
+{{--                                <a href="#" class="btn btn-warning">@lang('common.edit')</a>--}}
```
Comment (HDVinnie) [inline resources/views/Staff/poll/index.blade.php]: These commented out buttons can be removed.

Comment (gyakkun) [reply resources/views/Staff/poll/index.blade.php]: Done.

File: app/Http/Controllers/Staff/PollController.php [hunk 4]
```diff
@@ -99,4 +99,109 @@ public function store(StorePoll $request)
...
+    }
+
+    /**
+     * Delete A Poll.
+     *
+     * @param $id
+     *
+     * @return Illuminate\Http\RedirectResponse
+     */
+    public function destroy($id)
```
Comment (HDVinnie) [inline app/Http/Controllers/Staff/PollController.php]: Migrations need to be fixed as there is no delete cascade for poll options. There is only for poll votes. You could for now do these as we plan to refactor all migrations soon.
```
        $poll = Poll::with(['options'])->findOrFail($id);
        $poll->options->delete();
        $poll->delete();
```

**Comment by HDVinnie** [comment]
@gyakkun what's your thoughts on removing IPs altogether in this PR. See this issue it would close for ref. https://github.com/HDInnovations/UNIT3D-Community-Edition/issues/1114

**Comment by gyakkun** [comment]
> @gyakkun what's your thoughts on removing IPs altogether in this PR. See this issue it would close for ref. #1114

I share my opinion on removing ip_check with @HVRVKVT in #1114 . It's better to be done in this PR #1130 .

File: app/Http/Controllers/Staff/PollController.php (hunk 4 above)
Comment (gyakkun) [reply app/Http/Controllers/Staff/PollController.php]: Should be done with https://github.com/HDInnovations/UNIT3D-Community-Edition/pull/1126/commits/c6207c3f3b64acf5e08bce2e115e2d4c44b42f65. Reviews needed for sensitive db operation.

**Comment by gyakkun** [comment]
BTW I reckon it as a must for a private tracker to record user's ip, for auditing and cheat check.

The approch we adpot on a running PT in my organization is, adding sperated logging table. It should be configurable if the staff want the site running public.

**Comment by HDVinnie** [comment]
> BTW I reckon it as a must for a private tracker to record user's ip, for auditing and cheat check.
> 
> The approch we adpot on a running PT in my organization is, adding an independent ip logging table.
> 
> Still, the ip logging should be configurable if the staff want the site running public.

 IP Logging is 99% inefficient. I don't think its worth logging users IP addresses to catch the few that are to stupid or don't care enough to use a VPN...

**Comment by gyakkun** [comment]
> @gyakkun what's your thoughts on removing IPs altogether in this PR. See this issue it would close for ref. #1114

Removed in https://github.com/HDInnovations/UNIT3D-Community-Edition/pull/1126/commits/957ac57b4b5915462669b40139008c1ccf79f6f5 and https://github.com/HDInnovations/UNIT3D-Community-Edition/pull/1126/commits/41998673adb8e73d238f4d9902ad5cad305f45cc .

As it's fomerly a configurable option, remove it or not are both acceptable. All up to you and feel free to skip with cherry pick.

**Comment by HDVinnie** [comment]
> > @gyakkun what's your thoughts on removing IPs altogether in this PR. See this issue it would close for ref. #1114
> 
> Removed in [957ac57](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/957ac57b4b5915462669b40139008c1ccf79f6f5) and [4199867](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/41998673adb8e73d238f4d9902ad5cad305f45cc) .
> 
> As it's fomerly a configurable option, remove it or not are both acceptable. All up to you and feel free to skip with revert or cherry pick.

I feel its best to fully remove the option. You agree?

**Comment by gyakkun** [comment]
> > > @gyakkun what's your thoughts on removing IPs altogether in this PR. See this issue it would close for ref. #1114
> > 
> > 
> > Removed in [957ac57](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/957ac57b4b5915462669b40139008c1ccf79f6f5) and [4199867](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/41998673adb8e73d238f4d9902ad5cad305f45cc) .
> > As it's fomerly a configurable option, remove it or not are both acceptable. All up to you and feel free to skip with revert or cherry pick.
> 
> I feel its best to fully remove the option. You agree?

Agree.

**Comment by gyakkun** [comment]
> > > @gyakkun what's your thoughts on removing IPs altogether in this PR. See this issue it would close for ref. #1114
> > 
> > 
> > Removed in [957ac57](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/957ac57b4b5915462669b40139008c1ccf79f6f5) and [4199867](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/41998673adb8e73d238f4d9902ad5cad305f45cc) .
> > As it's fomerly a configurable option, remove it or not are both acceptable. All up to you and feel free to skip with revert or cherry pick.
> 
> I feel its best to fully remove the option. You agree?

What about replace it with "Result is only visible after votes?", it may be more useful.

**Comment by HDVinnie** [comment]
> > > > @gyakkun what's your thoughts on removing IPs altogether in this PR. See this issue it would close for ref. #1114
> > > 
> > > 
> > > Removed in [957ac57](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/957ac57b4b5915462669b40139008c1ccf79f6f5) and [4199867](https://github.com/HDInnovations/UNIT3D-Community-Edition/commit/41998673adb8e73d238f4d9902ad5cad305f45cc) .
> > > As it's fomerly a configurable option, remove it or not are both acceptable. All up to you and feel free to skip with revert or cherry pick.
> > 
> > 
> > I feel its best to fully remove the option. You agree?
> 
> What about replace it with "Result is only visible after votes?", it may be more useful.

I think the poll system can indeed use a lot more. Your idea is good. Others like this on polls table would be neat so you can close a poll at a certain date?:
```
            $table->boolean('closed')->default(false);
            $table->timestamp('closes_at')->nullable()->default(null);
```

Also on voters table it might be nice to add `option_id` so on the poll results page we can highlight what the said user voted for. If you know what I mean.

**Comment by gyakkun** [comment]
> > BTW I reckon it as a must for a private tracker to record user's ip, for auditing and cheat check.
> > The approch we adpot on a running PT in my organization is, adding an independent ip logging table.
> > Still, the ip logging should be configurable if the staff want the site running public.
> 
> IP Logging is 99% inefficient. I don't think its worth logging users IP addresses to catch the few that are to stupid or don't care enough to use a VPN...

I would better explain the necessity in a new open issue.  I think it's something more important than efficiency.

**Comment by torman365** [comment]
IP Tracking is a real necessity for any decent torrent tracker site despite VPN usage,

**Comment by gyakkun** [comment]
> I think the poll system can indeed use a lot more. Your idea is good. Others like this on polls table would be neat so you can close a poll at a certain date?:
> 
> ```
>             $table->boolean('closed')->default(false);
>             $table->timestamp('closes_at')->nullable()->default(null);
> ```
> 
> Also on voters table it might be nice to add `option_id` so on the poll results page we can highlight what the said user voted for. If you know what I mean.

So that can the style add to result page and make the voted option highlight according to the `option_id` , right? That sounds really nice. It would better be another story in another PR. Hope my PRs on poll feature won't bother your refactor plan, and I will move my hands off from it since I'm not so experienced 😂

Now things have all been done to remove ip_checking, and #1130  can be closed.

**Pull request merged**